.github/workflows/install_deps/action.yml

```diff
@@ -8,22 +8,32 @@ inputs:
     description: The version of Python to use in order to run unit tests
     required: false
     default: '3.9'
+  install_numpy_2:
+    description: Indicate if numpy 2.0+ should be installed or not
+    required: false
+    type: boolean
+    default: false
+  numpy_2_version:
+    description: The version of numpy 2.0 to use in order to run unit tests
+    required: false
+    type: string
+    default: '2.0.0'
   install_jax:
     description: Indicate if JAX should be installed or not
     required: false
     default: 'true'
   jax_version:
     description: The version of JAX to install for any job that requires JAX
     required: false
-    default: 0.4.23
+    default: '0.4.23'
```

Comment:
This will need to be 0.4.28, right?

Reply:
I believe this might be a different story: https://app.shortcut.com/xanaduai/story/61389/ensure-pennylane-is-compatible-with-jax-0-4-26

Reply:
Yep we should update this to 0.4.28 once the release is made (as Catalyst has been updated to work with this JAX version)

Reply:
Looks like updating JAX to 0.4.28 broke a lot of tests (thanks @mudit2812 for the [investigation](https://xanaduhq.slack.com/archives/C0271MF2M50/p1725474140063309)). At this point, we should probably open a second PR once this is merged, hopefully after [this PR](https://github.com/PennyLaneAI/pennylane/pull/6215) from @albi3ro is merged.